packages/client/src/pages/GameItem/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState } from "react";
+import { Header } from "components/Header";
+import styles from "./styles.module.scss";
+import { useParams, useHistory } from "react-router-dom";
+import { useApi } from "context/api";
+import { IApiError, IDiscount, IGame, IGameCreator } from "interfaces/api";
+import moment from "moment";
+import { IGameForUI } from "interfaces/app";
+import { Loader } from "components/Loader";
+import { getUserSessionData, setUserSessionData } from "utils/helpers";
+import { FaShoppingCart } from "react-icons/fa";
+import { useAuth } from "context/auth";
+
+interface GameItemProps {}
+
+export const GameItem: React.FunctionComponent<GameItemProps> = () => {
+  const { id } = useParams<{ id: string }>();
+  const { token } = useAuth();
+  const history = useHistory();
+  const {
+    getGame,
+    getUsedGenres,
+    getGenres,
+    getGameCreator,
+    getDiscounts,
+    getUsedDiscounts,
+  } = useApi();
+  const [error, setError] = useState<IApiError | null>(null);
+  const [game, setGame] = useState<IGameForUI | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [sessionData, setSessionData] = useState<number[]>([]);
+
+  const addToCartHandler = () => {
+    if (game && token) {
+      const sessionData = [...new Set([...getUserSessionData(), game.id])];
+      setUserSessionData(sessionData);
+      setSessionData(sessionData);
+    } else history.push("/login");
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (isNaN(id as any)) history.push("/store");
```

Comment:
Avoid type hacks at all costs. Especially with `any`. Also, checking `id` (which is string) with `isNaN()` seems pointless.